### ISSUE TITLE
esp8266 (NodeMCU V2 Amica, Wemos D1 mini): Can't connect to ESPHome API // Ping timed out!

### ISSUE THREAD

**Issue by luxing0815** [esphome/issues#3169]
### The problem

Since update to v2022.3.0 all my esp8266 have the problem that they are losing their connection to Home Assistant. With v2022.2.0 I didn't have any problems.

This is independent of their intended use and the connected components. For example, I use the esp8266 to control LED strips, as well as for garden irrigation, where they switch a 4-way relay only.

After restarting the system and rebooting the esp's, it takes only a few minutes until they mutually lose their connection and pings get lost on both sides.

I checked the history. The issue occurred immediately after the update. Until then there were only isolated connection problems. Maybe 2-5 connection issues per day. Now it's several hundred a day!

I have already restarted my wifi. And there are no abnormalities with other devices in my network. All my esp8266 have static ip and their wifi signal strength is -60db to -70db.

It might be the same issue as reported in #3154. However, the error for me is not related to any specific sensors.

### Which version of ESPHome has the issue?

2022.3.0

### What type of installation are you using?

Home Assistant Add-on

### Which version of Home Assistant has the issue?

2022.3.5

### What platform are you using?

ESP8266

### Board

NodeMCU V2 Amica, Wemos D1 mini

### Component causing the issue

api

### Example YAML snippet

```yaml
esphome:
  name: kitchen-rgbw-led-strip1
  platform: ESP8266
  board: nodemcuv2

# Enable logging
logger:
  level: VERBOSE

# Enable Home Assistant API
api:

ota:
  password: !secret ota_password

wifi:
  ssid: !secret wlan_ssid
  password: !secret wlan_password

  # Optional manual IP
  manual_ip:
    static_ip: 192.168.1.82
    gateway: 192.168.1.1
    subnet: 255.255.255.0

  # Enable fallback hotspot (captive portal) in case wifi connection fails
  ap:
    ssid: "Kitchen Led Strip 1 Hotspot"
    password: !secret ap_password

captive_portal:

globals:
  - id: my_global_int
    type: int
    restore_value: no
    initial_value: '0'

switch:
  - platform: template
    name: krls1_clr_forward
    lambda: |-
      return false;
    turn_on_action:
      - lambda: |-
          id(my_global_int) += 1;
          if (id(my_global_int) > 6) {
            id(my_global_int) = 1;
          }
          
          ESP_LOGD("krls1_clr_forward", "Global value is: %d", id(my_global_int));
          
          float r = 0.0, g = 0.0, b = 0.0;
          
          switch (id(my_global_int)) {
            case 1: r = 1.0; g = 0.0; b = 0.0; break;
            case 2: r = 0.5; g = 0.5; b = 0.0; break;
            case 3: r = 0.0; g = 1.0; b = 0.0; break;
            case 4: r = 0.0; g = 0.5; b = 0.5; break;
            case 5: r = 0.0; g = 0.0; b = 1.0; break;
            case 6: r = 0.5; g = 0.0; b = 0.5; break;
            default: break;
          }

          auto call = id(kitchen_rgbw_led_strip1).turn_on();
          call.set_transition_length(500);
          call.set_color_brightness(1.0);
          call.set_rgbw(r, g, b, 0.0);
          call.set_effect("None");
          call.perform();

  - platform: template
    name: krls1_clr_backward
    lambda: |-
      return false;
    turn_on_action:
      - lambda: |-
          id(my_global_int) = 0;

          ESP_LOGD("krls1_clr_backward", "Global value is: %d", id(my_global_int));
          
          auto call = id(kitchen_rgbw_led_strip1).turn_on();
          call.set_transition_length(500);
          call.set_color_brightness(0.0);
          call.set_white(1.0);
          call.set_effect("None");
          call.perform();

remote_receiver:
  pin:
    number: D2
    inverted: true
    mode: INPUT_PULLUP
  dump: lg

binary_sensor:
  - platform: remote_receiver
    name: krls1_ir_brighter
    lg:
      data: 0x00F700FF
      nbits: 32
    on_press:
      then:
       - light.dim_relative:
           id: kitchen_rgbw_led_strip1
           relative_brightness: 20%
    
  - platform: remote_receiver
    name: krls1_ir_darker
    lg:
      data: 0x00F7807F
      nbits: 32
    on_press:
      then:
       - light.dim_relative:
           id: kitchen_rgbw_led_strip1
           relative_brightness: -20%

  - platform: remote_receiver
    name: krls1_ir_off
    lg:
      data: 0x00F740BF
      nbits: 32
    on_press:
      then:
        - light.turn_off: kitchen_rgbw_led_strip1
    
  - platform: remote_receiver
    name: krls1_ir_on
    lg:
      data: 0x00F7C03F
      nbits: 32
    on_press:
      then:
        - light.turn_on: kitchen_rgbw_led_strip1
    
  - platform: remote_receiver
    name: krls1_ir_flash
    lg:
      data: 0x00F7D02F
      nbits: 32
    on_press:
      then:
        - light.turn_on: 
            id: kitchen_rgbw_led_strip1
            effect: pulse

  - platform: remote_receiver
    name: krls1_ir_strobe
    lg:
      data: 0x00F7F00F
      nbits: 32
    on_press:
      then:
        - light.turn_on: 
            id: kitchen_rgbw_led_strip1
            effect: strobe

  - platform: remote_receiver
    name: krls1_ir_fade
    lg:
      data: 0x00F7C837
      nbits: 32
    on_press:
      then:
        - light.turn_on: 
            id: kitchen_rgbw_led_strip1
            effect: random

  - platform: remote_receiver
    name: krls1_ir_smooth
    lg:
      data: 0x00F7E817
      nbits: 32
    on_press:
      then:
        - light.turn_on: 
            id: kitchen_rgbw_led_strip1
            effect: lambda

  - platform: remote_receiver
    name: krls1_ir_white
    lg:
      data: 0x00F7E01F
      nbits: 32
    on_press:
      then:
        - light.turn_on: 
            id: kitchen_rgbw_led_strip1
            color_brightness: 0%
            white: 100%
            red: 0%
            green: 0%
            blue: 0%
            effect: None

  - platform: remote_receiver
    name: krls1_ir_red
    lg:
      data: 0x00F720DF
      nbits: 32
    on_press:
      then:
        - light.turn_on: 
            id: kitchen_rgbw_led_strip1
            color_brightness: 100%
            white: 0%
            red: 100%
            green: 0%
            blue: 0%
            effect: None
        
  - platform: remote_receiver
    name: krls1_ir_green
    lg:
      data: 0x00F7A05F
      nbits: 32
    on_press:
      then:
        - light.turn_on: 
            id: kitchen_rgbw_led_strip1
            color_brightness: 100%
            white: 0%
            red: 0%
            green: 100%
            blue: 0%
            effect: None
            
  - platform: remote_receiver
    name: krls1_ir_blue
    lg:
      data: 0x00F7609F
      nbits: 32
    on_press:
      then:
        - light.turn_on: 
            id: kitchen_rgbw_led_strip1
            color_brightness: 100%
            white: 0%
            red: 0%
            green: 0%
            blue: 100%
            effect: None
            
  - platform: remote_receiver
    name: krls1_ir_orange
    lg:
      data: 0x00F710EF
      nbits: 32
    on_press:
      then:
        - light.turn_on: 
            id: kitchen_rgbw_led_strip1
            color_brightness: 100%
            white: 0%
            red: 100%
            green: 50%
            blue: 0%
            effect: None

  - platform: remote_receiver
    name: krls1_ir_light_green
    lg:
      data: 0x00F7906F
      nbits: 32
    on_press:
      then:
        - light.turn_on: 
            id: kitchen_rgbw_led_strip1
            color_brightness: 100%
            white: 0%
            red: 50%
            green: 100%
            blue: 0%
            effect: None

  - platform: remote_receiver
    name: krls1_ir_medium_blue
    lg:
      data: 0x00F750AF
      nbits: 32
    on_press:
      then:
        - light.turn_on: 
            id: kitchen_rgbw_led_strip1
            color_brightness: 100%
            white: 0%
            red: 0%
            green: 50%
            blue: 100%
            effect: None

  - platform: remote_receiver
    name: krls1_ir_light_orange
    lg:
      data: 0x00F730CF
      nbits: 32
    on_press:
      then:
        - light.turn_on: 
            id: kitchen_rgbw_led_strip1
            color_brightness: 100%
            white: 0%
            red: 100%
            green: 75%
            blue: 0%
            effect: None
  
  - platform: remote_receiver
    name: krls1_ir_light_blue
    lg:
      data: 0x00F7B04F
      nbits: 32
    on_press:
      then:
        - light.turn_on: 
            id: kitchen_rgbw_led_strip1
            color_brightness: 100%
            white: 0%
            red: 0%
            green: 100%
            blue: 100%
            effect: None

  - platform: remote_receiver
    name: krls1_ir_purple
    lg:
      data: 0x00F7708F
      nbits: 32
    on_press:
      then:
        - light.turn_on: 
            id: kitchen_rgbw_led_strip1
            color_brightness: 100%
            white: 0%
            red: 75%
            green: 0%
            blue: 100%
            effect: None

  - platform: remote_receiver
    name: krls1_ir_dark_yellow
    lg:
      data: 0x00F708F7
      nbits: 32
    on_press:
      then:
        - light.turn_on: 
            id: kitchen_rgbw_led_strip1
            color_brightness: 100%
            white: 0%
            red: 100%
            green: 82%
            blue: 28%
            effect: None

  - platform: remote_receiver
    name: krls1_ir_light_turquoise
    lg:
      data: 0x00F78877
      nbits: 32
    on_press:
      then:
        - light.turn_on: 
            id: kitchen_rgbw_led_strip1
            color_brightness: 100%
            white: 0%
            red: 28%
            green: 82%
            blue: 100%
            effect: None

  - platform: remote_receiver
    name: krls1_ir_light_purple
    lg:
      data: 0x00F748B7
      nbits: 32
    on_press:
      then:
        - light.turn_on: 
            id: kitchen_rgbw_led_strip1
            color_brightness: 100%
            white: 0%
            red: 82%
            green: 28%
            blue: 100%
            effect: None

  - platform: remote_receiver
    name: krls1_ir_yellow
    lg:
      data: 0x00F728D7
      nbits: 32
    on_press:
      then:
        - light.turn_on: 
            id: kitchen_rgbw_led_strip1
            color_brightness: 100%
            white: 0%
            red: 100%
            green: 100%
            blue: 0%
            effect: None

  - platform: remote_receiver
    name: krls1_ir_turquoise
    lg:
      data: 0x00F7A857
      nbits: 32
    on_press:
      then:
        - light.turn_on: 
            id: kitchen_rgbw_led_strip1
            color_brightness: 100%
            white: 0%
            red: 0%
            green: 75%
            blue: 100%
            effect: None

  - platform: remote_receiver
    name: krls1_ir_pink
    lg:
      data: 0x00F76897
      nbits: 32
    on_press:
      then:
        - light.turn_on: 
            id: kitchen_rgbw_led_strip1
            color_brightness: 100%
            white: 0%
            red: 100%
            green: 0%
            blue: 75%
            effect: None

output:
  - platform: esp8266_pwm
    id: krls1_pwm_white
    pin: D1
    max_power: 100%

  - platform: esp8266_pwm
    id: krls1_pwm_red
    pin: D5
    max_power: 100%

  - platform: esp8266_pwm
    id: krls1_pwm_green
    pin: D6
    max_power: 100%

  - platform: esp8266_pwm
    id: krls1_pwm_blue
    pin: D7
    max_power: 100%

light:
  - platform: rgbw
    name: "Kitchen RGBW LED Strip 1"
    id: kitchen_rgbw_led_strip1
    white: krls1_pwm_white
    red: krls1_pwm_red
    green: krls1_pwm_green
    blue: krls1_pwm_blue
    effects:
      - pulse:
          transition_length: 0.5s
          update_interval: 0.5s
      - strobe:
          colors:
            - state: true
              brightness: 100%
              red: 100%
              green: 90%
              blue: 0%
              duration: 500ms
            - state: false
              duration: 250ms
            - state: true
              brightness: 100%
              red: 0%
              green: 100%
              blue: 0%
              duration: 500ms
      - random:
          transition_length: 4s
          update_interval: 5s
      - lambda:
          name: Custom
          update_interval: 1s
          lambda: |-
            static int state = 0;
            auto call = id(kitchen_rgbw_led_strip1).turn_on();
            // Transition of 1000ms = 1s
            call.set_transition_length(1000);
            if (state == 0) {
              call.set_rgbw(.0, .0, .0, 1.0);
            } else if (state == 1) {
              call.set_rgbw(1.0, .0, 1.0, .0);
            } else if (state == 2) {
              call.set_rgbw(.0, .0, 1.0, .0);
            } else {
              call.set_rgbw(1.0, .0, .0, .0);
            }
            call.perform();
            state += 1;
            if (state == 4)
              state = 0;
```

### Anything in the logs that might be useful for us?

```txt
iot device:
===========
[08:33:40][I][app:102]: ESPHome version 2022.3.0 compiled on Mar 21 2022, 17:26:11
[08:33:40][C][wifi:491]: WiFi:
[08:33:40][C][wifi:353]:   Local MAC: F4:CF:A2:EC:09:EC
[08:33:40][C][wifi:354]:   SSID: [redacted]
[08:33:40][C][wifi:355]:   IP Address: 192.168.1.82
[08:33:40][C][wifi:356]:   BSSID: [redacted]
[08:33:40][C][wifi:358]:   Hostname: 'kitchen-rgbw-led-strip1'
[08:33:40][C][wifi:360]:   Signal strength: -61 dB ▂▄▆█
...
[08:33:41][V][mdns:086]:   Services:
[08:33:41][V][mdns:088]:   - _esphomelib, _tcp, 6053
[08:33:41][V][mdns:090]:     TXT: version = 2022.3.0
[08:33:41][V][mdns:090]:     TXT: mac = f4cfa2ec09ec
[08:33:41][V][mdns:090]:     TXT: platform = ESP8266
[08:33:41][V][mdns:090]:     TXT: board = nodemcuv2
[08:33:41][C][ota:085]: Over-The-Air Updates:
[08:33:41][C][ota:086]:   Address: 192.168.1.82:8266
[08:33:41][C][ota:089]:   Using Password.
[08:33:41][C][api:138]: API Server:
[08:33:41][C][api:139]:   Address: 192.168.1.82:6053
[08:33:41][C][api:143]:   Using noise encryption: NO
INFO 192.168.1.82: Ping timed out!
INFO Disconnected from ESPHome API for 192.168.1.82
WARNING Disconnected from API
INFO Successfully connected to 192.168.1.82
[08:34:46][D][api:102]: Accepted 192.168.1.80
[08:34:46][V][api.connection:809]: Hello from client: 'Home Assistant 2022.3.5 (192.168.1.80)'
[08:34:46][D][api.connection:827]: Home Assistant 2022.3.5 (192.168.1.80): Connected successfully
INFO 192.168.1.82: Ping timed out!
INFO Disconnected from ESPHome API for 192.168.1.82
WARNING Disconnected from API
WARNING Can't connect to ESPHome API for 192.168.1.82: Timeout while connecting to ('192.168.1.82', 6053)
INFO Trying to reconnect to 192.168.1.82 in the background
INFO Successfully connected to 192.168.1.82
INFO 192.168.1.82: Ping timed out!
INFO Disconnected from ESPHome API for 192.168.1.82
WARNING Disconnected from API
INFO Successfully connected to 192.168.1.82
[08:36:33][D][api:102]: Accepted 192.168.1.80
[08:36:33][V][api.connection:809]: Hello from client: 'Home Assistant 2022.3.5 (192.168.1.80)'
[08:36:33][D][api.connection:827]: Home Assistant 2022.3.5 (192.168.1.80): Connected successfully
[08:36:33][V][api.connection:893]: Cannot send message because of TCP buffer space
[08:36:33][V][api.connection:893]: Cannot send message because of TCP buffer space
[08:36:33][V][api.connection:893]: Cannot send message because of TCP buffer space
[08:36:33][V][api.connection:893]: Cannot send message because of TCP buffer space
...
```

### Additional information

home assistant:
==========
2022-03-22 08:33:27 INFO (MainThread) [aioesphomeapi.reconnect_logic] Successfully connected to kitchen-rgbw-led-strip1 @ 192.168.1.82
2022-03-22 08:34:08 INFO (MainThread) [aioesphomeapi.connection] kitchen-rgbw-led-strip1 @ 192.168.1.82: Ping timed out!
2022-03-22 08:34:08 INFO (MainThread) [aioesphomeapi.reconnect_logic] Disconnected from ESPHome API for kitchen-rgbw-led-strip1 @ 192.168.1.82
2022-03-22 08:34:38 WARNING (MainThread) [aioesphomeapi.reconnect_logic] Can't connect to ESPHome API for kitchen-rgbw-led-strip1 @ 192.168.1.82: Timeout while connecting to ('192.168.1.82', 6053)
2022-03-22 08:34:38 INFO (MainThread) [aioesphomeapi.reconnect_logic] Trying to reconnect to kitchen-rgbw-led-strip1 @ 192.168.1.82 in the background
2022-03-22 08:34:46 INFO (MainThread) [aioesphomeapi.reconnect_logic] Successfully connected to kitchen-rgbw-led-strip1 @ 192.168.1.82
2022-03-22 08:35:29 INFO (MainThread) [aioesphomeapi.connection] kitchen-rgbw-led-strip1 @ 192.168.1.82: Ping timed out!
2022-03-22 08:35:29 INFO (MainThread) [aioesphomeapi.reconnect_logic] Disconnected from ESPHome API for kitchen-rgbw-led-strip1 @ 192.168.1.82
2022-03-22 08:35:29 INFO (MainThread) [aioesphomeapi.reconnect_logic] Successfully connected to kitchen-rgbw-led-strip1 @ 192.168.1.82
2022-03-22 08:36:26 INFO (MainThread) [aioesphomeapi.connection] kitchen-rgbw-led-strip1 @ 192.168.1.82: Ping timed out!
2022-03-22 08:36:26 INFO (MainThread) [aioesphomeapi.reconnect_logic] Disconnected from ESPHome API for kitchen-rgbw-led-strip1 @ 192.168.1.82
2022-03-22 08:36:33 INFO (MainThread) [aioesphomeapi.reconnect_logic] Successfully connected to kitchen-rgbw-led-strip1 @ 192.168.1.82
2022-03-22 08:36:38 WARNING (MainThread) [homeassistant.components.esphome] Error getting initial data for 192.168.1.82: Timeout waiting for response for <class 'api_pb2.ListEntitiesRequest'>
2022-03-22 08:36:48 INFO (MainThread) [aioesphomeapi.reconnect_logic] Disconnected from ESPHome API for kitchen-rgbw-led-strip1 @ 192.168.1.82
2022-03-22 08:36:48 INFO (MainThread) [aioesphomeapi.connection] kitchen-rgbw-led-strip1 @ 192.168.1.82: Ping Failed: Socket closed while reading data: 0 bytes read on a total of 1 expected bytes
2022-03-22 08:37:03 INFO (MainThread) [aioesphomeapi.reconnect_logic] Successfully connected to kitchen-rgbw-led-strip1 @ 192.168.1.82
2022-03-22 08:38:13 INFO (MainThread) [aioesphomeapi.connection] kitchen-rgbw-led-strip1 @ 192.168.1.82: Ping timed out!
2022-03-22 08:38:13 INFO (MainThread) [aioesphomeapi.reconnect_logic] Disconnected from ESPHome API for kitchen-rgbw-led-strip1 @ 192.168.1.82
2022-03-22 08:38:21 INFO (MainThread) [aioesphomeapi.reconnect_logic] Successfully connected to kitchen-rgbw-led-strip1 @ 192.168.1.82
2022-03-22 08:40:51 INFO (MainThread) [aioesphomeapi.connection] kitchen-rgbw-led-strip1 @ 192.168.1.82: Ping timed out!
2022-03-22 08:40:51 INFO (MainThread) [aioesphomeapi.reconnect_logic] Disconnected from ESPHome API for kitchen-rgbw-led-strip1 @ 192.168.1.82
2022-03-22 08:40:51 INFO (MainThread) [aioesphomeapi.reconnect_logic] Successfully connected to kitchen-rgbw-led-strip1 @ 192.168.1.82
2022-03-22 08:42:06 INFO (MainThread) [aioesphomeapi.connection] kitchen-rgbw-led-strip1 @ 192.168.1.82: Ping timed out!
2022-03-22 08:42:06 INFO (MainThread) [aioesphomeapi.reconnect_logic] Disconnected from ESPHome API for kitchen-rgbw-led-strip1 @ 192.168.1.82
2022-03-22 08:42:06 INFO (MainThread) [aioesphomeapi.reconnect_logic] Successfully connected to kitchen-rgbw-led-strip1 @ 192.168.1.82
2022-03-22 08:42:51 INFO (MainThread) [aioesphomeapi.connection] kitchen-rgbw-led-strip1 @ 192.168.1.82: Ping timed out!
2022-03-22 08:42:51 INFO (MainThread) [aioesphomeapi.reconnect_logic] Disconnected from ESPHome API for kitchen-rgbw-led-strip1 @ 192.168.1.82
2022-03-22 08:42:52 INFO (MainThread) [aioesphomeapi.reconnect_logic] Successfully connected to kitchen-rgbw-led-strip1 @ 192.168.1.82
2022-03-22 08:43:32 INFO (MainThread) [aioesphomeapi.connection] kitchen-rgbw-led-strip1 @ 192.168.1.82: Ping timed out!
2022-03-22 08:43:32 INFO (MainThread) [aioesphomeapi.reconnect_logic] Disconnected from ESPHome API for kitchen-rgbw-led-strip1 @ 192.168.1.82
2022-03-22 08:44:02 WARNING (MainThread) [aioesphomeapi.reconnect_logic] Can't connect to ESPHome API for kitchen-rgbw-led-strip1 @ 192.168.1.82: Timeout while connecting to ('192.168.1.82', 6053)
2022-03-22 08:44:02 INFO (MainThread) [aioesphomeapi.reconnect_logic] Trying to reconnect to kitchen-rgbw-led-strip1 @ 192.168.1.82 in the background
2022-03-22 08:44:11 INFO (MainThread) [aioesphomeapi.reconnect_logic] Successfully connected to kitchen-rgbw-led-strip1 @ 192.168.1.82
2022-03-22 08:44:36 INFO (MainThread) [aioesphomeapi.connection] kitchen-rgbw-led-strip1 @ 192.168.1.82: Ping timed out!
2022-03-22 08:44:36 INFO (MainThread) [aioesphomeapi.reconnect_logic] Disconnected from ESPHome API for kitchen-rgbw-led-strip1 @ 192.168.1.82
2022-03-22 08:44:36 INFO (MainThread) [aioesphomeapi.reconnect_logic] Successfully connected to kitchen-rgbw-led-strip1 @ 192.168.1.82
2022-03-22 08:46:00 INFO (MainThread) [aioesphomeapi.connection] kitchen-rgbw-led-strip1 @ 192.168.1.82: Ping timed out!
2022-03-22 08:46:00 INFO (MainThread) [aioesphomeapi.reconnect_logic] Disconnected from ESPHome API for kitchen-rgbw-led-strip1 @ 192.168.1.82
2022-03-22 08:46:00 INFO (MainThread) [aioesphomeapi.reconnect_logic] Successfully connected to kitchen-rgbw-led-strip1 @ 192.168.1.82
2022-03-22 08:46:30 INFO (MainThread) [aioesphomeapi.connection] kitchen-rgbw-led-strip1 @ 192.168.1.82: Error while reading incoming messages: Error while reading data: [Errno 104] Connection reset by peer
2022-03-22 08:46:30 INFO (MainThread) [aioesphomeapi.reconnect_logic] Disconnected from ESPHome API for kitchen-rgbw-led-strip1 @ 192.168.1.82
2022-03-22 08:46:30 INFO (MainThread) [aioesphomeapi.connection] kitchen-rgbw-led-strip1 @ 192.168.1.82: Ping Failed: Error while reading data: [Errno 104] Connection reset by peer
2022-03-22 08:46:30 INFO (MainThread) [aioesphomeapi.reconnect_logic] Successfully connected to kitchen-rgbw-led-strip1 @ 192.168.1.82
2022-03-22 08:47:28 INFO (MainThread) [aioesphomeapi.reconnect_logic] Successfully connected to kitchen-rgbw-led-strip1 @ 192.168.1.82

**Comment by luxing0815** [comment]
Apparently this is a known issue. A workaround can be found in #1237, which in turn refers to #1532.
If have added **power_save_mode: high** to the wifi section for all of esp8266. This really makes this difference. I can confirm what was observed by @CarlosGS in #issuecomment-718669014.

**Comment by luxing0815** [comment]
If have added a binary ping sensor for each of my esp8266 with a scan interval of 60s. Now, with **power_save_mode: high**, whithin 24 hrs I have counted roughly 50 disconnects but only one warning ("Can't connect to ESPHome API")  per device. That's quite an improvement.
Whereby I have update my system to Home Assistant 2022.3.6 in the middle of this time period. But there was no noticable difference. The history graph looks pretty much the same before and after.
Fortunately, the remaining disconnections were not noticeable, as all esp8266 responded quickly and reliably to all my automation triggers. At least for now. 
I will continue to monitor my system...

**Comment by luxing0815** [comment]
Update to Home Assistant 2022.3.7 and esphome 2022.3.1 did not bring any negative surprises. After a restart, the system runs very smoothly. Only once the error "Can't connect to ESPHome API" within a day. 
So, the workaround (power_save_mode: high) has solved my problem, which is why close this ticket now.

**Comment by luxing0815** [comment]
Reopen because, unfortunately, not solved.

Within the past 72 hours the system has reported again dozens of "Can't connect to ESPHome API" warnings. I have added now a sensor to record the wifi signal strengh. Here are my current measurements:

Drip irrigation system (signal strength: -80db +/- 2db)
8x Can't connect to ESPHome API ...: Error connecting to
19x Can't connect to ESPHome API ...: Timeout while connecting to

LED Strip 1 (signal strength: -66db +/- 2db)
3x Can't connect to ESPHome API ...: Error connecting to
0x Can't connect to ESPHome API ...: Timeout while connecting

LED Strip 2 (signal strength: -70db +/- 2db)
17x Can't connect to ESPHome API ...: Error connecting to
20x Can't connect to ESPHome API ...: Timeout while connecting to

**Comment by CarlosGS** [comment]
Hi! Seeing `"Cannot send message because of TCP buffer space"` makes me wonder if it is an issue with low RAM.
Could you try commenting out `captive_portal:`? I think this would free up some memory.
Also maybe remove `level: VERBOSE` from the logger component, it sometimes can cause problems. Good luck!

**Comment by luxing0815** [comment]
> Hi! Seeing `"Cannot send message because of TCP buffer space"` makes me wonder if it is an issue with low RAM. Could you try commenting out `captive_portal:`? I think this would free up some memory. Also maybe remove `level: VERBOSE` from the logger component, it sometimes can cause problems. Good luck!

Thank you for the hint. I have used `level: VERBOSE` only for my root cause analysis. And you are right that this probably has caused the `Cannot send message because of TCP buffer space` error.
For field use, however, I am using the default log level (`level: WARN`). The memory usage is 40% RAM and 43% Flash. Commenting out `captive_portal` reduces the of usage each by ~2,5%. 
The disconnect of the esp8266 continues to take place.

**Comment by luxing0815** [comment]
I evaluated the Wifi sensors over a longer period of time and found that the signal strength of all three esp8266 fluctuates at the same time (up to +/- 5 db). Then I scanned the wifi frequencies in different places in the house and found six wifi routers on "my" channel with a similarly high signal strength. Fiber to the Home was recently rolled out here and now all of our neighbors probably have the same router running on the same channel. After I switched all esp8266 to another, more powerful wifi access point on a different channel, I don't have a single disconnect anymore. The esp8266 respond very quickly. That's how it should be. I hereby close this ticket.
